zk/lock.go

```diff
@@ -72,9 +83,9 @@ func (l *Lock) Lock() error {
 					return err
 				}
 			}
-		} else if err == nil {
+		case nil:
```

Comment:
this won't work the way it was intendet. 
break will only break switch statement.

switch should be removed or you need to add loop point

```
loop:
for i := 0; i < 3; i++ {
   switch:
   case nil:
      break loop
}
```
Example:
https://goplay.x1unix.com/snippet/xL1H5y8AvXB

Reply:
Hi Metalscreame,
I have not touched the original code. This structure has been in existence for some time now.

Reply:
Hi Nikhil. But there were `if else` structure and you changed it with `switch-case` statement

Reply:
Apologies Metalscreame! I realized it later on.

Reply:
Thanks for pointing out. It somehow got missed from me. Please recheck